public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -148,21 +163,43 @@ function AddAnomalyDetector({ embeddable, closeFlyout, mode, setMode }) {
     }
   };
 
-  const handleSubmit = (formikProps) => {
+  const getAugmentVisSavedObject = (detectorId: string) => {
+    const fn = {
+      type: VisLayerTypes.PointInTimeEvents,
+      name: OVERLAY_ANOMALIES,
+      args: {
+        detectorId: detectorId,
+      },
+    } as VisLayerExpressionFn;
+
+    const pluginResource = {
+      type: VIS_LAYER_PLUGIN_TYPE,
+      id: detectorId,
+    } as ISavedPluginResource;
+
+    return {
+      title: embeddable.vis.title,
+      originPlugin: ORIGIN_PLUGIN_VIS_LAYER,
+      pluginResource: pluginResource,
+      visId: embeddable.vis.id,
+      visLayerExpressionFn: fn,
+    } as ISavedAugmentVis;
```

Comment:
nit: we will want a `pluginEventType` field (+ probably a new constant stored here). This is added in [this PR](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4123)

Reply:
I only see this being added to PointInTimeEventsVisLayer, not ISavedAugmentVis

Reply:
Ah sorry, yes you're right